src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java

```diff
@@ -526,15 +528,20 @@ public RESTEnkelvoudigInformatieobject updateEnkelvoudigInformatieobject(
                         zrcClientService.readZaak(enkelvoudigInformatieObjectVersieGegevens.zaakUuid)
                 ).wijzigen()
         );
-        final var file = (RESTFileUpload) httpSession.get()
-                .getAttribute("FILE_" + enkelvoudigInformatieObjectVersieGegevens.zaakUuid);
-        var updatedDocument = informatieobjectConverter.convert(enkelvoudigInformatieObjectVersieGegevens, file);
-        updatedDocument = enkelvoudigInformatieObjectUpdateService.updateEnkelvoudigInformatieObjectWithLockData(
-                parseUUIDFromResourceURI(document.getUrl()),
-                updatedDocument,
-                enkelvoudigInformatieObjectVersieGegevens.toelichting
-        );
-        return informatieobjectConverter.convertToREST(convertToEnkelvoudigInformatieObject(updatedDocument));
+        final var file = data != null ? data : (RESTFileUpload) httpSession.get().getAttribute("FILE_" +
```

Comment:
Static string should be declared as a constant.
And we could use String Templates from [Java 21 preview features](https://docs.oracle.com/en/java/javase/21/language/string-templates.html#GUID-78F545D3-CDD0-415C-9B4B-6EF361D184F5)...

Reply:
Hmm, please no more preview features.. I really don't like to use Java preview features. I think they are pain because you need to configure that --enable-preview JVM flag everywhere. And every developers needs to do that. And even then you are never 100% sure that feature will make it into Java for real. I am glad we finally got rid of those preview features now. :-)

Reply:
Extracted as constant